File: df4k-core/src/main/java/org/df4j/core/util/invoker/ConsumerInvoker.java
package org.df4j.core.util.invoker;

import java.util.function.Consumer;

public class ConsumerInvoker<U,R> extends AbstractInvoker<Consumer<U>, R> {

    public ConsumerInvoker(Consumer<U> consumer) {
        super(consumer);
    }

    public R apply(Object... args) {
        assert args.length == 1;
        function.accept((U) args[0]);
        return null;
    }

}

File: df4k-core/src/main/kotlin/org/df4k/core/util/invoker/ConsumerInvoker.kt
package org.df4k.core.util.invoker

import java.util.function.Consumer

class ConsumerInvoker<U, R>(consumer: Consumer<U>) : AbstractInvoker<Consumer<U>, R>(consumer) {

    override fun apply(vararg args: Any): R? {
        assert(args.size == 1)
        function!!.accept(args[0] as U)
        return null
    }

}
